### PR TITLE
API: direct submission

### PR DESCRIPTION
**Pull request by terceiro**

### CODE REVIEW

**Comment by terceiro** [comment]
@mwasilew need your feedback. in special on the filenames of the test result files.

**Comment by terceiro** [comment]
I kind of understand why the test is failing on travis, will look tomorrow

File: benchmarks/testminer.py [hunk 1]
```diff
@@ -112,10 +112,39 @@ def __init__(self, base_url, username, password):
...
+                self.url,
+                username=self.username,
+                password=self.password
+            )
+        return self.__jenkins__
+
+    @property
+    def jenkins_job(self):
+        if self.__jenkins_job__ is None:
+            self.__jenkins_job__ = self.jenkins[self.job_name]
```
Comment (mwasilew) [inline benchmarks/testminer.py]: For some reason this fails on jenkinsapi 0.2.32 (at least it fails for me). I didn't debug further. The solution was to downgrade to 0.2.30

File: api/tests.py [hunk 2]
```diff
@@ -366,6 +368,38 @@ def test_add_existing_test_jobs_to_existing_result_object(self):
...
         self.assertEqual(models.TestJob.objects.count(), 2)
 
+    @patch.dict('django.conf.settings.CREDENTIALS', {'jenkins.linaro.org': ("hej", "ho")})
+    def test_post_already_with_testjob_results(self):
+        data = {
+            'build_url': 'http://jenkins.linaro.org/foo/bar/baz/1',
+            'name': u'linaro-art-stable-m-build-juno',
+            'url': u'http://dynamicfixture1.com',
+            'build_number': 200,
+            'build_id': 20,
```
Comment (mwasilew) [inline api/tests.py]: I think build_number should be the same as build_id. I'm not sure why there are 2 fields for this property.

File: api/views.py [hunk 3]
```diff
@@ -232,10 +233,36 @@ def create(self, request, *args, **kwargs):
...
+
+                self.__create_test_job__(result, env, filedata)
 
         tasks.update_jenkins.delay(result)
         return response.Response(serializer.data, status=status.HTTP_201_CREATED)
 
+    def __create_test_job__(self, result, env, data):
+        environment = benchmarks_models.Environment.objects.get_or_create(identifier=env)
+        testjob = benchmarks_models.TestJob.objects.create(
+            id='J' + str(result.build_id) + '_' + result.name + '+' + env,
```
Comment (mwasilew) [inline api/views.py]: why '+' not '_'?

File: api/views.py [hunk 4]
```diff
@@ -232,10 +233,36 @@ def create(self, request, *args, **kwargs):
...
+            result=result,
+            status='Complete',
+            initialized=True,
+            completed=True,
+            data=data,
+            testrunnerclass='ArtJenkinsTestResults',
+            testrunnerurl='http://example.com/', # FIXME
```
Comment (mwasilew) [inline api/views.py]: testrunnerurl = result.build_url

**Comment by mwasilew** [comment]
This looks OK. I also assumed that filename of the .json == environment name. This was requested from Vishal and it's implemented in the uBenchmark build jobs now.

**Comment by mwasilew** [comment]
I think the failure is caused by StringIO in JSON. Any submission fails for me locally.

**Comment by mwasilew** [comment]
Got it working locally. Looks good. Merging.

**Pull request merged**